### PR TITLE
Sample and command line refactor

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Refactors samples not to rely on `cmdUtils` to setup MQTT clients, to connect/disconnect for connect samples, and overall just makes `cmdUtils` go back to command line parsing and nothing more.

Also contains the following minor changes:
* Renames CustomKeyOpsPubSub to CustomKeyOpsConnect, removes PubSub part of sample
  * This is because the only difference this sample shows is the connection method, therefore no reason to include the PubSub part.
* Renames `Identity` to `FleetProvisioning` for better visibility. Added note both in README and sample itself that `Identity` = `FleetProvisioning`
* Removed `PubSubStress` sample entirely
* Adjusts all samples to register all command line commands and get the data from said commands in a single function, rather than doing it individually in each sample.
  * This also fixed a few minor inconsistencies that slipped through: like `ca` being used in a couple samples instead of `ca_file`.
* Fixes possible Cross-SDK repository communication for PubSub samples
  * Shadow still has potential to communicate across SDKs, as it uses a classic (unnamed) shadow currently.
* Removed RawConnect sample entirely
  * The AWS IoT MQTT Builder should be used instead. If a sample on using the CRT is wanted/needed, it should be in the CRT repository.
* Trimmed `import` calls in all samples to just those that are needed.
__________

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by xiazhvera — APPROVED** [review]
The change looks good to me, and I think this should be the way to go. 

While I think Thoai had comments about we should have a more clear main function body. As an example, we can have a static function to contains the build steps in the sample file instead of the util command, so that the main function body only contains the simple function call as before. 

It would be good to have more team member look at it.

File: samples/Mqtt5/SharedSubscription/src/main/java/sharedsubscription/SharedSubscription.java [hunk 1]
```diff
@@ -189,61 +189,43 @@ public static SampleMqtt5Client createMqtt5Client(
...
+         * cmdData is the arguments/input from the command line placed into a single struct for
+         * use in this sample. This handles all of the command line parsing, validating, etc.
+         * See the Utils/CommandLineUtils for more information.
+         */
+        CommandLineUtils.SampleCommandLineData cmdData = CommandLineUtils.getInputForIoTSample("Mqtt5SharedSubscription", args);
 
         /* If this is CI, append a UUID to the topic */
         if (isCI) {
-            input_topic += "/" + UUID.randomUUID().toString();
+            cmdData.input_topic += "/" + UUID.randomUUID().toString();
```
Comment (xiazhvera) [inline samples/Mqtt5/SharedSubscription/src/main/java/sharedsubscription/SharedSubscription.java:201]: Can we make the appending an UUID to the topic as a default behavior like the client id? Would we have the similar CI problem in other samples? 
```
returnData.input_clientId = getCommandOrDefault(m_cmd_client_id, "test-" + UUID.randomUUID().toString());
```

Comment (TwistedTwigleg) [reply samples/Mqtt5/SharedSubscription/src/main/java/sharedsubscription/SharedSubscription.java:201]: Good idea - I'll move it to the `getCommandOrDefault` for the topic!

**Comment by TwistedTwigleg** [comment]
Update 04/07/2023: Removed the `RawConnect` sample, as we want customers to use the AWS IoT MQTT builders we provide for easy access to AWS. If something like RawConnect is wanted/needed, we should add it to the CRT repository (`aws-crt-java`).

**Comment by TwistedTwigleg** [comment]
Thanks! Merging into `main`...

**Pull request merged**